Update APM Server documentation for use since 8.x requiring Kibana. (#7559)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/apm-server.asciidoc

[id="{p}-apm-eck-managed-es"]
== Use an Elasticsearch cluster managed by ECK

> Managing the APM Server a Kibana instance and Elasticsearch with ECK allows a smooth and secured integration between the stack components. The output configuration of the APM Server is setup automatically to establish a trust relationship with Elasticsearch. Specifying the Kibana reference allows ECK to automatically configure the Kibana endpoint as described in https://www.elastic.co/guide/en/apm/server/current/setup-kibana-endpoint.html

Reviewer: I find this first sentence a bit hard to parse. Not sure if I have a better suggestion though. 
```suggestion
Managing APM Server, Kibana and Elasticsearch with ECK allows a smooth and secured integration between the stack components. The output configuration of the APM Server is setup automatically to establish a trust relationship with Elasticsearch. Specifying the Kibana reference allows ECK to automatically configure the  link:https://www.elastic.co/guide/en/apm/server/current/setup-kibana-endpoint.html[Kibana endpoint].
```

  count: 1
  elasticsearchRef:
    name: quickstart
  kibanaRef: <1>
    name: quickstart
EOF
----

<1> A reference to a Kibana instance is only required for APM Server versions 8.0.0 and later.

> NOTE: If Kibana is referenced in the APM Server specification then the following configuration is required in the Kibana specification to install the APM ILM templates:

Reviewer: Is that true? It is only required if you are using version 8.0.0 and later. So maybe: something along the lines of: "starting with version 8.0.0 the following Kibana configuration is requireed to run APM Server"

Reviewer: Yes but my comment was more about the wording you had. It implies that this is required whenever a `kibanaRef` is specified, when in fact we should qualify this with "starting with 8.0.0"

Author: @pebrc My comment may have been misleading, as I had also updated the wording, in additional to double checking the requirement.

EOF
----

> You will also need to ensure that your Kibana instance has the following configuration:

Reviewer: ```suggestion
Starting with version 8.0.0 you will also need to ensure that your Kibana instance has the following configuration:
```
